mm2src/coins/utxo/utxo_common.rs

```diff
@@ -3298,8 +3298,25 @@ pub fn min_tx_amount(coin: &UtxoCoinFields) -> BigDecimal {
 }
 
 pub fn min_trading_vol(coin: &UtxoCoinFields) -> MmNumber {
-    let dust_multiplier = MmNumber::from(10);
-    dust_multiplier * min_tx_amount(coin).into()
+    let min_from_dust = coin.dust_amount * 10;
+
+    let fixed_fee_rate = match coin.tx_fee {
+        FeeRate::FixedPerKb(sats) => Some(ActualFeeRate::FixedPerKb(sats)),
+        FeeRate::FixedPerKbDingo(sats) => Some(ActualFeeRate::FixedPerKbDingo(sats)),
+        // For coins with dynamic fees, an async call would be needed to get the current fee rate.
+        // Since this function is synchronous and its output must be deterministic for the lifetime of
+        // an order / swap, we will use the dust-based calculation.
+        FeeRate::Dynamic(_) => None,
```

Comment:
I think this alone is a good enough justification. I wouldn't avoid doing dynamic calculation just because the function isn't async.

```suggestion
        // The output of this function must be deterministic for the lifetime of
        // an order / swap, we will use the dust-based calculation.
        FeeRate::Dynamic(_) => None,
```